gen.go

```diff
@@ -37,9 +66,18 @@ func doTemplate(w io.Writer, info interface{}, templ string) error {
 			"ReadHeader": func(rdr string) string {
 				return fmt.Sprintf(`%s.ReadHeader()`, rdr)
 			},
-			"MaxLen": func(val int, def string) string {
+			"MaxLen": func(val int, defaultType string) string {
```

Comment:
In retrospect, we probably could have just gone with a number (enum) for `defaultType`.

Reply:
Yeah, but this comes in from a template so we'd end up having to use integers there which is more opaque `{{ MaxLen .MaxLen "String" }}` -> `{{ MaxLen .MaxLen 2 }}`. Unless we share them as functions that take no arguments, but then we'd have to pipe: `{{ String | MaxLen .MaxLen }}`. Unless I'm missing a template feature that could be used here?